paramak/parametric_shapes/sweep_circle_shape.py

```diff
@@ -0,0 +1,123 @@
+from collections import Iterable
+
+import cadquery as cq
+
+from paramak import Shape
+
+
+class SweepCircleShape(Shape):
+    """Sweeps a 2D circle of a defined radius along a defined spline path to create a
+    3D CadQuery solid.
+
+    Args:
+        radius (float): Radius of 2D circle to be swept.
+        path_points (list of tuples each containing X (float), Z (float)): A list of XY,
+            YZ or XZ coordinates connected by spline connections which define the path
+            along which the 2D shape is swept.
+        workplane (str): Workplane in which the circle to be swept is defined.
+        path_workplane (str): Workplane in which the spline path is defined.
+    """
```

Comment:
Here some non-default arguments appear to be missing (stp_filename and stl_filename) as well as default values for workplane and path_workplane